### DISCUSSION TITLE
Autofill persists even though I am turning it off

### DISCUSSION THREAD

**Discussion by alaahariri** [dani-garcia/vaultwarden#4983] [Q&A] [answered]
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.32.0
* Web-vault version: v2024.6.2b
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.46.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** DOMAIN, DISABLE_ICON_DOWNLOAD, SIGNUPS_ALLOWED, SIGNUPS_VERIFY, SIGNUPS_DOMAINS_WHITELIST, INVITATIONS_ALLOWED, SHOW_PASSWORD_HINT, ADMIN_TOKEN, INVITATION_ORG_NAME, SMTP_HOST, SMTP_SECURITY, SMTP_PORT, SMTP_FROM, SMTP_USERNAME, SMTP_PASSWORD, SMTP_AUTH_MECHANISM

```json
{
  "_duo_akey": null,
  "_enable_duo": true,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_max_note_size": 10000,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://**************",
  "domain_origin": "*****://**************",
  "domain_path": "",
  "domain_set": true,
  "duo_context_purge_schedule": "30 * * * * *",
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "duo_use_iframe": false,
  "email_2fa_auto_fallback": false,
  "email_2fa_enforce_on_verified_invite": false,
  "email_attempts_limit": 3,
  "email_change_allowed": false,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "enable_websocket": false,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "http_request_block_non_global_ips": true,
  "http_request_block_regex": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "increase_note_size_limit": false,
  "invitation_expiration_hours": 120,
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": "data/vaultwarden.log",
  "log_level": "error",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "****",
  "org_events_enabled": false,
  "org_groups_enabled": true,
  "password_hints_allowed": true,
  "password_iterations": 600000,
  "push_enabled": false,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "********",
  "signups_verify": true,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": "\"Login\"",
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "**************",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "*********************",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "**************",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

### Vaultwarden Build Version

v1.32.0

### Deployment method

Official Container Image

### Reverse Proxy

nginx 1.26

### Host/Server Operating System

Linux

### Operating System Version

RHEL 9.4

### Clients

Web Vault, Browser Extension

### Client Version

Edge: Version: 2024.9.1

### Steps To Reproduce

1. Go to Browser Icon
2. Click on Settings
3. Click on Autofill
4. Click on Show autofill menu on form fields
5. Select Off

The browser keeps showing the icon/menu on any website form fields.

### Expected Result

Vaultwarden icon / dropdown list shouldn't be shown.

### Actual Result

The browser keeps showing the autofill icon/menu.

It should not display it.

**Comment by BlackDex** [comment]
This looks like a client issue. Vaultwarden does not maintain or develop any Bitwarden client.
If this is a an issue then i suggest to check the Bitwarden Community forums or there GitHub issue pages.

**Reply by alaahariri** [reply]
Thank you @BlackDex 

When switching to accounts hosted on Bitwarden.com, the autofill feature got disabled as expected. However, the issue only occurs for Vaultwarden self-hosted accounts.

I even tried removing and reinstalling the Bitwarden extension, but that didn't resolve the problem.

**Reply by BlackDex** [reply]
Well, it can't be a Vaultwarden issue, as those details are not stored on the server side. This is all a client side functionality.

**Reply by alaahariri** [reply]
Check this Env template https://github.com/dani-garcia/vaultwarden/blob/d184c8f08cfdbc567d47d8edd473b8d22ebe8bd9/.env.template#L347 

There seem to be some sever side settings that control autofill?

Also, this discussion is referring to the same https://github.com/dani-garcia/vaultwarden/discussions/4254#discussioncomment-8171522

**Reply by BlackDex** [reply]
That has been removed since v2024.2.0 already. So that feature flag will do nothing at all anymore.

**Reply by alaahariri** [reply]
That's weird! 

Thank you for the info and guidance.

**Reply by bendem** [reply]
It looks like people reporting the problem upstream are all self hosted. Are you sure it's not a new feature? Something that would sync client settings to the server?

https://github.com/bitwarden/clients/issues/11173

**Reply by BlackDex** [reply]
If it is a new feature, then the clients either should respect the feature flags if that is used, else the default should work out of the box which then probably is wrong right now.

If also the Bitwarden Self Hosted is causing this issue there is not much we can do really. I have no clue which feature flag would solve/trigger this.

**Comment by Gerardv514** [comment]
Does this happen for all entries or some?

**Reply by alaahariri** [reply]
**Let me brief you below about my experience in this bug:** 

**Client Details;** 
- MS Edge: Version 129.0.2792.52 (Official build) (64-bit)
- Bitwarden Edge Extension: Version: 2024.9.1
- Bitwarden client is running with two Accounts (one from Bitwarden.com and one from self-hosted Vaultwarden).
- Autofill is turned off from Bitwarden extension settings. 

**Scenarios:** 
- Activate Bitwarden.com account: Form fields from any website doesn't show any Bitwarden overlays (Icon or Dropdown) as per the chosen settings.
- Activate Self-hosted account and refresh the webpage: Form fields on any site show the icon and overlay on the fields. Thus, it doesn't respect the extension settings.
- Re-switch back to the Bitwarden account, refresh the page, web form fields show nothing as it should be.

**Reply by Gerardv514** [reply]
I can say I am not seeing the same behavior in my self hosted instance, however I use chrome. 

Can you try chrome browser and confirm? Maybe this is an issue with edge browser extension?

**Reply by alaahariri** [reply]
I have just installed the Chrome latest version, installed Bitwarden extension, and the same issue occurs. I just want to point out that only the Off option
![image](https://github.com/user-attachments/assets/561b77c7-e3ed-4275-aba8-4a72f14f8520)

is not working. The other two options (overlay and icon only) are behaving correctly.

**Reply by alaahariri** [reply]
Could you please share your Vaultwarden Support String? I would be comparing them with ours.

**Reply by Gerardv514** [reply]
> I have just installed the Chrome latest version, installed Bitwarden extension, and the same issue occurs. I just want to point out that only the Off option ![image](https://private-user-images.githubusercontent.com/17312481/370304502-561b77c7-e3ed-4275-aba8-4a72f14f8520.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcxODcwNDAsIm5iZiI6MTcyNzE4Njc0MCwicGF0aCI6Ii8xNzMxMjQ4MS8zNzAzMDQ1MDItNTYxYjc3YzctZTNlZC00Mjc1LWFiYTgtNGE3MmYxNGY4NTIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI0VDE0MDU0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZiZTViMTgxNDQxZWViOWI5ZjY5N2Q3ZDEzMzc3NTc4MjE2NTllYjc5NDQ1YTU5NzE5MTMyYmMwYTEyZDA3NzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.KyfPo44B8Tvqe5Z2S9GIzJNjPfUVemXJliS6R2JPO8Y)
> 
> is not working. The other two options (overlay and icon only) are behaving correctly.

Can you show a screenshot of a login page with this happening?

**Comment by dfunkt** [comment, marked as answer]
This is a problem with the web extension and it has already been reported to the Bitwarden devs: https://github.com/bitwarden/clients/issues/11173
I have also encountered it, with both Vaultwarden and Bitwarden.

**Reply by gcstr** [reply]
Apparently, [they solved it](https://github.com/bitwarden/clients/issues/11173#issuecomment-2374834245), but even after updating Vaultwarden, I can still see the problem.

**Reply by BlackDex** [reply]
The clients have nothing to do with Vaultwarden regarding these features.
So the correlation is invalid in this case.

**Reply by dfunkt** [reply]
@gcstr: Make sure your extension version is 2024.10.0 or higher.
That's all it took to fix the issue for me.
